tests/integration_tests/databases/api_tests.py

```diff
@@ -3619,3 +3619,45 @@ def test_validate_sql_endpoint_failure(self, get_validator_by_name):
             return
         self.assertEqual(rv.status_code, 422)
         self.assertIn("Kaboom!", response["errors"][0]["message"])
+
+    @mock.patch(
+        "superset.security.SupersetSecurityManager.get_schemas_accessible_by_user"
+    )
+    @mock.patch("superset.security.SupersetSecurityManager.can_access_database")
+    @mock.patch("superset.security.SupersetSecurityManager.can_access_all_datasources")
+    def test_schemas_access_for_csv_upload_not_found_endpoint(
+        self,
+        mock_can_access_all_datasources,
+        mock_can_access_database,
+        mock_schemas_accessible,
+    ):
+        self.login(username="gamma")
+        self.create_fake_db()
+        mock_can_access_all_datasources.return_value = False
+        mock_can_access_database.return_value = False
+        mock_schemas_accessible.return_value = ["this_schema_is_allowed_too"]
+        rv = self.client.get(f"/api/v1/database/120ff/schemas_access_for_file_upload")
+        self.assertEqual(rv.status_code, 404)
+        self.delete_fake_db()
```

Comment:
can you add another simple test to assert a database id that does not exist returns a 404 from this endpoint?